test/put.js

```diff
@@ -174,51 +175,49 @@ describe('client.put()', function () {
     var record = rgen(key, meta)
 
     // write the record and then check
-    client.put(key, record, meta, function (err, key) {
-      expect(err).to.be.ok()
-      expect(err.code).to.equal(status.AEROSPIKE_OK)
+    Aerospike.put(key, record, meta, function (err, key, status) {
+      expect(err).not.to.be.ok()
 
-      client.get(key, function (err, record1, metadata, key) {
-        if (err && err.code !== status.AEROSPIKE_OK) { throw new Error(err.message) }
+      Aerospike.get(key, function (err, record1, metadata, key) {
+        if (err) { throw new Error(err.message) }
         expect(record1).to.eql(record)
 
-        client.remove(key, function (err, key) {
-          if (err && err.code !== status.AEROSPIKE_OK) { throw new Error(err.message) }
+        Aerospike.remove(key, function (err, key) {
+          if (err) { throw new Error(err.message) }
           done()
         })
       })
     })
   })
 
-  it('shoule write a bin with double value', function (done) {
-    // generators
-    var kgen = keygen.string(options.namespace, options.set, {prefix: 'test/put/'})
-    var mgen = metagen.constant({ttl: 1000})
-
-    // values
-    var key = kgen()
-    var meta = mgen(key)
-    var record = {
-      val: 123.45,
-      dval: Double(456.00)
-    }
-    // write the record and then check
-    client.put(key, record, meta, function (err, key) {
-      expect(err).to.be.ok()
-      expect(err.code).to.equal(status.AEROSPIKE_OK)
-
-      client.get(key, function (err, record1, metadata, key) {
-        if (err && err.code !== status.AEROSPIKE_OK) { throw new Error(err.message) }
-        expect(record1.val).to.equal(record.val)
-        expect(record1.dval).to.equal(record.dval.Double)
-
-        client.remove(key, function (err, key) {
-          if (err && err.code !== status.AEROSPIKE_OK) { throw new Error(err.message) }
-          done()
-        })
-      })
-    })
-  })
+  // it('shoule write a bin with double value', function (done) {
+  //   // generators
+  //   var kgen = keygen.string(options.namespace, options.set, {prefix: 'test/put/'})
+  //   var mgen = metagen.constant({ttl: 1000})
+  //
+  //   // values
+  //   var key = kgen()
+  //   var meta = mgen(key)
+  //   var record = {
+  //     val: 123.45,
+  //     dval: Double(456.00)
+  //   }
+  //   // write the record and then check
+  //   Aerospike.put(key, record, meta, function (err, key) {
+  //     expect(err).not.to.be.ok()
+  //
+  //     Aerospike.get(key, function (err, record1, metadata, key) {
+  //       if (err) { throw new Error(err.message) }
+  //       expect(record1.val).to.equal(record.val)
+  //       expect(record1.dval).to.equal(record.dval.Double)
+  //
+  //       Aerospike.remove(key, function (err, key) {
+  //         if (err) { throw new Error(err.message) }
+  //         done()
+  //       })
+  //     })
+  //   })
+  // })
```

Comment:
What server version are you testing against? Support for double data type was introduced in server version 3.6. Older servers don't handle unsupported data types well and will crash if they receive double values.

You can send an info request and check that the `features` field contains an entry `float`.